src/Elastic.Apm/Report/PayloadSenderV2.cs

```diff
@@ -73,27 +74,30 @@ public PayloadSenderV2(
 		)
 			: base(isEnabled, logger, ThisClassName, service, configuration, httpMessageHandler)
 		{
-			if (!isEnabled)
-				return;
-
 			_logger = logger?.Scoped(ThisClassName + (dbgName == null ? "" : $" (dbgName: `{dbgName}')"));
+			_isEnabled = isEnabled;
+			if (!_isEnabled)
+				_logger?.Debug()?.Log($"{nameof(PayloadSenderV2)} is not enabled, work loop won't be started.");
+
 			_payloadItemSerializer = new PayloadItemSerializer();
 			_configuration = configuration;
 			_brokenActivationMethodVersion = new ElasticVersion(8, 7, 0);
-
 			_intakeV2EventsAbsoluteUrl = BackendCommUtils.ApmServerEndpoints.BuildIntakeV2EventsAbsoluteUrl(configuration.ServerUrl);
 
 			System = system;
 
 			_cloudMetadataProviderCollection = new CloudMetadataProviderCollection(configuration.CloudProvider, _logger, environmentVariables);
 			_apmServerInfo = apmServerInfo ?? new ApmServerInfo();
 			_serverInfoCallback = serverInfoCallback;
+
 			var process = ProcessInformation.Create();
 			_metadata = new Metadata { Service = service, System = System, Process = process };
 			foreach (var globalLabelKeyValue in configuration.GlobalLabels)
 				_metadata.Labels.Add(globalLabelKeyValue.Key, globalLabelKeyValue.Value);
 			_cachedActivationMethod = _metadata.Service?.Agent.ActivationMethod;
-			ResetActivationMethodIfKnownBrokenApmServer();
+
+			if (isEnabled)
```

Comment:
nit: for consistency, should we prefer the `_isEnabled` field rather than the parameter here? We use the parameter above.